app/controllers/quizzes_controller.rb

```diff
@@ -50,25 +54,33 @@ def update
 
   def submit
     @quiz = Quiz.find(params[:id])
-    @quiz_session = QuizSession.find_by!(user: current_user, quiz: @quiz, state: "in_progress")
-    result = []
-    params[:questions].each do |question_param|
-      question = Question.find_by(id: question_param[:id], quiz_id: @quiz.id)
-      if question.nil?
-        result << {
-          id: question_param[:id],
-          status: "Error; Question not found"
-        }
-      else
-        result << {
-          id: question.id,
-        }.merge(question.check(question_param))
+    @quiz_session = QuizSession.find_by(user: current_user, quiz: @quiz, state: "in_progress")
+    if @quiz_session.nil?
+      render json: {
+        error: "No attempts left!"
```

Comment:
avem o metoda mai buna pt erori in ApplicationController
```
return render_error(
      status: :not_found,
      code: "no_session_found"
    )
```
Si pt ca dai `return` nu mai tb sa folosesti `else` la linia 62